### DISCUSSION TITLE
posts/wake-on-lan/

### DISCUSSION THREAD

**Discussion by giscus** [techno-tim/techno-tim.github.io#233] [Comments]
# posts/wake-on-lan/

What is Wake on LAN and why is it so hard?

https://technotim.live/posts/wake-on-lan/

**Comment by jacobnollette** [comment]
Any possibility to sleep/wake linux vm's?

**Reply by timothystewart6** [reply]
I am not sure however I would probably just use the API from the Hypervisor to do this. e.g. if you're using proxmox, either use their REST API or Ansible / CLI to turn on one of the machines.

**Comment by michaelpetschel** [comment]
Does this work on a raspberry pi

**Reply by timothystewart6** [reply]
You can send the packets from a Pi, but not wake a Pi, it does not support it.

**Comment by wakawakaaa** [comment]
Hi Tim, 
I tried but when 2 LAN Ports are connected as a Bond, it dont work. Do you have any solution for it?

Bond mode is LACP (802.3ad) and vmbr0 is a Linux Bridge.

Thank for help.

**Comment by LessTaken** [comment]
Hey Tim, 
I found your guide while googeling to set up WoL for the ZimaBlade.

Technical requirements should be all fullfilled, BIOS settings are enabled, WoL service was active anyways. But I still struggle to wake the board up again. In my understanding, all LAN activity would wake up the machine, like an SSH command, or storage access. Or am I wrong?

**Reply by LessTaken** [reply]
Edit: I found my issue, sorry for the post...

For everyone else with the same question: you have to send a "magic" packet to your ethernet port mac adress.
I found this tool for windows: https://www.heise.de/download/product/wol.exe-43799/download
Just use it in Powershell like "wol.exe <MACWithoutAnySeperator>

For Ubuntu e.g. this can be done with: sudo etherwake <mac-adresse> -i eth0

**Comment by pnotsam** [comment]
I suspect a power plug with a nice web interface like Tasmota for each computer may be a way to avoid all this complexity and allow any device to be simply switched on or off.  For convenience powerplugs can be added to HomeAssistant on a dashboard to easily power cycle any frozen equipment.  It could also be configured to powerup the Ezcoo and PiKVM only when needed.  Trying to reconcile several different implemtations of WoL in Bios, Operating systems and varieties of NiC channel bonding and switches seems like a difficult balancing act and requires a lot of time to get a working reliable combination. 

C'est manifique! Mais c'est pas du guerre!

**Comment by Raagam-Parmar** [comment]
Thanks a lot! I was upset thinking this wouldn't work on my TrueNAS Scale setup, but it did! Thank you very much !

**Comment by skaczmarek-dev** [comment]
You don't have to use wakeonlan with sudo.

**Comment by pixelnovae** [comment]
thanks for the great guide Tim. a follow up question:

the set up i am working with: mac mini (sonoma), connected to network via ethernet to the wireless modem and macbook (sonoma), connected wireless

goal: wake up (turn on?) the mac mini from a distance with my macbook

so far: i have set up all the power settings on the mac mini as prescribed, installed "wakeonlan" via brew on both the mini and the macbook, and when sending the "wake [mini mac address]" from my macbook, nothing happens. any insights as to why?

the dumb question: can one wake up a mac minin that has been "shut down"

**Comment by Jibun-no-Kage** [comment]
My new Mac Mini M4 Pro does not respond to WOL.  I have validated the configuration on the Mac, and tried WOL from Linux as well as Windows.  My Windows and Linux systems that support WOL work fine.  But not the Mac.  Never mind that the Mac goes not support power states comparable to X86_64 hardware which is frustrating, that the Mac never come out of power off comparable to other hardware... but even when the Mac is in sleep mode, does not respond to WOL at all now.

**Comment by hungry-bogart** [comment]
I think I'm having an issue with persistence on this setting. I followed the systemd setup and it has been working the last few days, but all of a sudden it stopped working. I have to investigate, but I was just curious if anyone else had similar trouble.

**Reply by hungry-bogart** [reply]
Ok. Here's a easy fix. Make sure your VPN is off. I also noted that one of my servers was only responsive if a packet was sent to a specific interface. So, if you have two interfaces, consider that a thing to check for.
